waspc/src/Wasp/Generator/ServerGenerator/Auth/LocalAuthG.hs

```diff
@@ -20,33 +24,42 @@ import qualified Wasp.Generator.AuthProviders.Local as Local
 import Wasp.Generator.FileDraft (FileDraft)
 import Wasp.Generator.Monad (Generator)
 import qualified Wasp.Generator.ServerGenerator.Common as C
+import Wasp.Generator.ServerGenerator.JsImport (extImportToImportJson)
 import qualified Wasp.Util as Util
 
 genLocalAuth :: AS.Auth.Auth -> Generator [FileDraft]
-genLocalAuth auth
-  | AS.Auth.isUsernameAndPasswordAuthEnabled auth =
-      sequence
-        [ genLoginRoute auth,
-          genSignupRoute auth,
-          genLocalAuthConfig
-        ]
-  | otherwise = return []
+genLocalAuth auth = case usernameAndPasswordAuth of
+  Just usernameAndPasswordAuthConfig ->
+    sequence
+      [ genLocalAuthConfig usernameAndPasswordAuthConfig,
+        genLoginRoute auth,
+        genSignupRoute auth
+      ]
+  Nothing -> return []
+  where
+    usernameAndPasswordAuth = AS.Auth.usernameAndPassword $ AS.Auth.methods auth
 
-genLocalAuthConfig :: Generator FileDraft
-genLocalAuthConfig = return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplData)
+genLocalAuthConfig :: AS.Auth.UsernameAndPasswordConfig -> Generator FileDraft
+genLocalAuthConfig usernameAndPasswordConfig = return $ C.mkTmplFdWithDstAndData tmplFile dstFile (Just tmplData)
   where
     tmplFile = C.srcDirInServerTemplatesDir </> SP.castRel authIndexFileInSrcDir
     dstFile = C.serverSrcDirInServerRootDir </> authIndexFileInSrcDir
 
     tmplData =
       object
         [ "providerId" .= Local.providerId localAuthProvider,
-          "displayName" .= Local.displayName localAuthProvider
+          "displayName" .= Local.displayName localAuthProvider,
+          "userFieldsFn" .= extImportToImportJson relPathToServerSrcDir maybeGetUserFieldsFn
         ]
 
+    maybeGetUserFieldsFn = AS.Auth.getUsernameAndPasswordUserFieldsFn usernameAndPasswordConfig
+
     authIndexFileInSrcDir :: Path' (Rel C.ServerSrcDir) File'
     authIndexFileInSrcDir = [relfile|auth/providers/config/username.ts|]
 
+    relPathToServerSrcDir :: Path Posix (Rel importLocation) (Dir C.ServerSrcDir)
+    relPathToServerSrcDir = [reldirP|../../../|]
```

Comment:
How did you determine this relative path? Could you maybe explain that in the comment, would that be helpful?

Reply:
This is established pattern in our codebase by now, this is the relative path from the import location to the server src dir. We have it in 3+ places and we didn't add any special comments above, so I would avoid putting it here as well.

Reply:
Hm ok, although I don't really get what it is. Is this relative part to server src dir from the file that we are generating? What does importLocation mean?
My main question is: in which cases can this path become incorrect? And can we do something about it, be it a comment or code design?

Reply:
Let's deal with these relative import issues when we solve the general issue e.g. introducing aliases https://github.com/wasp-lang/wasp/issues/1011 or rewriting our ext import implementation.